warehouse/integrations/snowflake/snowflake.go

```diff
@@ -397,49 +476,157 @@ func (sf *Snowflake) loadTable(tableName string, tableSchemaInUpload model.Table
-		additionalJoinClause = fmt.Sprintf(`AND original."%[1]s" = staging."%[1]s" AND original."%[2]s" = staging."%[2]s"`, "TABLE_NAME", "COLUMN_NAME")
+		additionalJoinClause = fmt.Sprintf(`AND original.%[1]q = staging.%[1]q AND original.%[2]q = staging.%[2]q`, "TABLE_NAME", "COLUMN_NAME")
 	}
 
-	keepLatestRecordOnDedup := sf.Uploader.ShouldOnDedupUseNewRecord()
-
 	if keepLatestRecordOnDedup {
-		sqlStatement = fmt.Sprintf(`MERGE INTO %[9]s."%[1]s" AS original
-									USING (
-										SELECT * FROM (
-											SELECT *, row_number() OVER (PARTITION BY %[8]s ORDER BY RECEIVED_AT DESC) AS _rudder_staging_row_number FROM %[9]s."%[2]s"
-										) AS q WHERE _rudder_staging_row_number = 1
-									) AS staging
-									ON (original."%[3]s" = staging."%[3]s" %[7]s)
-									WHEN MATCHED THEN
-									UPDATE SET %[6]s
-									WHEN NOT MATCHED THEN
-									INSERT (%[4]s) VALUES (%[5]s)`, tableName, stagingTableName, primaryKey, sortedColumnNames, stagingColumnNames, columnsWithValues, additionalJoinClause, partitionKey, schemaIdentifier)
+		sqlStatement = fmt.Sprintf(`
+			MERGE INTO %[9]s.%[1]q AS original USING (
+			  SELECT
+				*
+			  FROM
+				(
+				  SELECT
+					*,
+					row_number() OVER (
+					  PARTITION BY %[8]s
+					  ORDER BY
+						RECEIVED_AT DESC
+					) AS _rudder_staging_row_number
+				  FROM
+					%[9]s.%[2]q
+				) AS q
+			  WHERE
+				_rudder_staging_row_number = 1
+			) AS staging ON (
+			  original.%[3]q = staging.%[3]q %[7]s
+			)
+			WHEN MATCHED THEN
+			  UPDATE SET %[6]s
+			WHEN NOT MATCHED THEN
+			  INSERT (%[4]s) VALUES (%[5]s);
+`,
+			tableName,
+			stagingTableName,
+			primaryKey,
+			sortedColumnNames,
+			stagingColumnNames,
+			columnsWithValues,
+			additionalJoinClause,
+			partitionKey,
+			schemaIdentifier,
+		)
 	} else {
-		sqlStatement = fmt.Sprintf(`MERGE INTO %[8]s."%[1]s" AS original
-										USING (
-											SELECT * FROM (
-												SELECT *, row_number() OVER (PARTITION BY %[7]s ORDER BY RECEIVED_AT DESC) AS _rudder_staging_row_number FROM %[8]s."%[2]s"
-											) AS q WHERE _rudder_staging_row_number = 1
-										) AS staging
-										ON (original."%[3]s" = staging."%[3]s" %[6]s)
-										WHEN NOT MATCHED THEN
-										INSERT (%[4]s) VALUES (%[5]s)`, tableName, stagingTableName, primaryKey, sortedColumnNames, stagingColumnNames, additionalJoinClause, partitionKey, schemaIdentifier)
+		sqlStatement = fmt.Sprintf(`
+			MERGE INTO %[8]s.%[1]q AS original USING (
+			  SELECT
+				*
+			  FROM
+				(
+				  SELECT
+					*,
+					row_number() OVER (
+					  PARTITION BY %[7]s
+					  ORDER BY
+						RECEIVED_AT DESC
+					) AS _rudder_staging_row_number
+				  FROM
+					%[8]s.%[2]q
+				) AS q
+			  WHERE
+				_rudder_staging_row_number = 1
+			) AS staging ON (
+			  original.%[3]q = staging.%[3]q %[6]s
+			)
+			WHEN NOT MATCHED THEN
+			  INSERT (%[4]s) VALUES (%[5]s)
+			WHEN MATCHED THEN
+			  UPDATE SET original.received_at = original.received_at;
```

Comment:
**Adding a dummy update block. We are just updating the received_at with its original values, not the staging values.**